### PR TITLE
read in vs to density scaling from file

### PR DESCRIPTION
**Pull request by jaustermann**
I've introduced an option to read in a vs to density scaling from a file. This is relevant for S40RTS and SAVANI. For now I've only included this in S40RTS but can update SAVANI accordingly once I've gotten feedback on this pull request.

### CODE REVIEW

File: include/aspect/initial_temperature/S40RTS_perturbation.h [hunk 1]
```diff
@@ -83,6 +88,20 @@ namespace aspect
...
+         * An enum to describe where the depth dependency of the viscosity is coming from.
+         */
+        enum VsToDensityMethod
+        {
+          File,
+          Constant
+        };
+
+        /**
+         * Currently chosen source for the viscosity.
```
Comment (gassmoeller) [inline include/aspect/initial_temperature/S40RTS_perturbation.h]: not a viscosity

File: include/aspect/initial_temperature/S40RTS_perturbation.h [hunk 2]
```diff
@@ -83,6 +88,20 @@ namespace aspect
 
       private:
 
+        /**
+         * An enum to describe where the depth dependency of the viscosity is coming from.
```
Comment (gassmoeller) [inline include/aspect/initial_temperature/S40RTS_perturbation.h]: Not a viscosity

File: include/aspect/initial_temperature/S40RTS_perturbation.h [hunk 3]
```diff
@@ -151,6 +170,16 @@ namespace aspect
          */
         std::shared_ptr<internal::S40RTS::SplineDepthsLookup> spline_depths_lookup;
 
+        /**
+         * Object containing the data profile.
+         */
+        aspect::Utilities::AsciiDataProfile<dim> profile;
+
+        /**
+         * The column indix of the vs to density scaling in the data file
```
Comment (gassmoeller) [inline include/aspect/initial_temperature/S40RTS_perturbation.h]: index

File: include/aspect/initial_temperature/S40RTS_perturbation.h [hunk 4]
```diff
@@ -83,6 +88,20 @@ namespace aspect
 
       private:
 
+        /**
+         * An enum to describe where the depth dependency of the viscosity is coming from.
+         */
+        enum VsToDensityMethod
+        {
+          File,
+          Constant
```
Comment (gassmoeller) [inline include/aspect/initial_temperature/S40RTS_perturbation.h]: we usually do not capitalize the options in an enum

File: source/initial_temperature/S40RTS_perturbation.cc [hunk 5]
```diff
@@ -279,11 +291,24 @@ namespace aspect
       // Get value at specific depth
       const double perturbation = s(scoord[0]);
 
+      // Get the vs to density converstion
+      const double depth = this->get_geometry_model().depth(position);
+
+      double vs_to_density = 0.0;
+      if ( vs_to_density_method == File)
```
Comment (gassmoeller) [inline source/initial_temperature/S40RTS_perturbation.cc]: please be consistent with the spaces after parentheses

File: source/initial_temperature/S40RTS_perturbation.cc [hunk 6]
```diff
@@ -316,6 +341,10 @@ namespace aspect
                              Patterns::Anything(),
                              "The file name of the spline knot locations from "
                              "Ritsema et al.");
+          prm.declare_entry ("Vs to density scaling method", "Constant",
```
Comment (gassmoeller) [inline source/initial_temperature/S40RTS_perturbation.cc]: same here, values of parameters are usually not capitalized (except for names)

File: tests/initial_condition_vs_to_density.prm [hunk 7]
```diff
@@ -0,0 +1,91 @@
+set Dimension                              = 3
```
Comment (gassmoeller) [inline tests/initial_condition_vs_to_density.prm]: Please add a comment what this test is for

**Comment by jaustermann** [comment]
@gassmoeller I've addressed your comments and fixed the test. Could you take another look?

**Comment by jaustermann** [comment]
I would also like to implement this in SAVANI. Should I do it within this pull request or submit a new pull request?

**Comment by bangerth** [comment]
If you promise to do it, then do it in a separate PR. This way we can already merge this one and it is not blocked by the functionality you'd like to add.

**Comment by jaustermann** [comment]
Okay! One test failed because a typo in the test parameter file. Fixed it, let's see if it passes now.

**Comment by jaustermann** [comment]
@gassmoeller these tests passed - could you take another look? Let me know if the commits are okay or if I should condense (squash) them.

**Comment by gassmoeller** [comment]
Looks good! Thanks for addressing the comments. :+1:

**Pull request merged**